dotnet/src/IntegrationTests/Planners/Stepwise/FunctionCallingStepwisePlannerTests.cs

```diff
@@ -0,0 +1,124 @@
+            .Build();
+
+        string? bingApiKeyCandidate = this._configuration["Bing:ApiKey"];
+        Assert.NotNull(bingApiKeyCandidate);
+        this._bingApiKey = bingApiKeyCandidate;
+    }
+
+    [Theory(Skip = "Requires model deployment that supports function calling.")]
+    [InlineData("What is the tallest mountain on Earth? How tall is it?", "Everest", new string[] { "WebSearch_Search" })]
+    [InlineData("What is the weather in Seattle?", "Seattle", new string[] { "WebSearch_Search" })]
+    [InlineData("What is the current hour number, plus 5?", "", new string[] { "Time_HourNumber", "Math_Add" })]
+    [InlineData("What is 387 minus 22? Email the solution to John and Mary.", "365", new string[] { "Math_Subtract", "Email_GetEmailAddress", "Email_SendEmail" })]
+    public async Task CanExecuteStepwisePlanAsync(string prompt, string partialExpectedAnswer, string[] expectedFunctions)
+    {
+        // Arrange
+        bool useEmbeddings = false;
+        Kernel kernel = this.InitializeKernel(useEmbeddings);
+        var bingConnector = new BingConnector(this._bingApiKey);
+        var webSearchEnginePlugin = new WebSearchEnginePlugin(bingConnector);
+        kernel.ImportPluginFromObject(webSearchEnginePlugin, "WebSearch");
+        kernel.ImportPluginFromType<TimePlugin>("Time");
+        kernel.ImportPluginFromType<MathPlugin>("Math");
+        kernel.ImportPluginFromType<EmailPluginFake>("Email");
+
+        var planner = new FunctionCallingStepwisePlanner(
+            new FunctionCallingStepwisePlannerConfig() { MaxIterations = 10 });
+
+        // Act
+        var planResult = await planner.ExecuteAsync(kernel, prompt);
+
+        // Assert - should contain the expected answer & function calls within the maximum iterations
+        Assert.NotNull(planResult);
+        Assert.NotEqual(string.Empty, planResult.FinalAnswer);
+        Assert.True(planResult.Iterations > 0);
+        Assert.True(planResult.Iterations <= 10);
+        Assert.Contains(partialExpectedAnswer, planResult.FinalAnswer, StringComparison.InvariantCultureIgnoreCase);
+
+        string serializedChatHistory = JsonSerializer.Serialize(planResult.ChatHistory);
+        foreach (string expectedFunction in expectedFunctions)
+        {
+            Assert.Contains(expectedFunction, serializedChatHistory, StringComparison.InvariantCultureIgnoreCase);
+        }
+    }
+
+    private Kernel InitializeKernel(bool useEmbeddings = false)
+    {
+        AzureOpenAIConfiguration? azureOpenAIConfiguration = this._configuration.GetSection("AzureOpenAI").Get<AzureOpenAIConfiguration>();
+        Assert.NotNull(azureOpenAIConfiguration);
+
+        IKernelBuilder builder = Kernel.CreateBuilder()
+            .AddAzureOpenAIChatCompletion(
+                deploymentName: azureOpenAIConfiguration.ChatDeploymentName!,
+                endpoint: azureOpenAIConfiguration.Endpoint,
+                apiKey: azureOpenAIConfiguration.ApiKey);
+
+        if (useEmbeddings)
```

Comment:
nit: Looks like new test never uses embeddings, so we can remove this flag and condition here.